src/main/java/com/cinema/controllers/ShoppingCartController.java

```diff
@@ -36,15 +38,17 @@ public ShoppingCartController(UserService userService,
 
     @PostMapping("/movie-sessions")
     public void addMovieSession(@RequestParam Long movieSessionId,
-                                @RequestParam Long userId) {
+                                Authentication authentication) {
         MovieSession movieSession = movieSessionService.get(movieSessionId);
-        User user = userService.get(userId);
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+        User user = userService.findByEmail(userDetails.getUsername()).get();
         shoppingCartService.addSession(movieSession, user);
     }
 
     @GetMapping("/by-user")
-    public ShoppingCartResponseDto getShoppingCartByUser(@RequestParam Long userId) {
-        User user = userService.get(userId);
+    public ShoppingCartResponseDto getShoppingCartByUser(Authentication authentication) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
```

Comment:
you can use authentication.getName() as well here

Reply:
agree. used in shopping cart controller